### ISSUE TITLE
Get YouTube data from API, not page scrape?

### ISSUE THREAD

**Issue by morbus** [rg3/youtube-dl#376]
G'day. I've been working on some local scripts to archive YouTube videos, and part of that is adding in metadata to the downloaded mp4 files from youtube-dl. After noticing that the description was coming out truncated (caused by me missing lxml), I'm now noticing that youtube-dl only knows the day (YYYY-MM-DD) and not the time the video was uploaded. This is through no fault of the script, however, as there is actually no complete timestamp on the video page itself.

The complete timestamp DOES seem to appear in the API though:
https://gdata.youtube.com/feeds/api/videos/WN5IzJ99Qys?v=2

I'm not a Python coder, but I could probably figure out to get this support in, but the real question is: would you be OK with youtube-dl making an additional request per video URL? (If not, that's fine too, as I'd just write a custom API script to fetch the data and ignore the --write-info-json option.)

**Comment by BrainStone** [comment]
I know this issue is a whopping 8 years old, but I noticed the Data API is still not being used so I set out to find out why.  
And it seems this issue gives the answer.

Now checking the current terms I cannot find anything that would indicate using it in the way youtube-dl works as a violation. I might have missed something though, but I think a reevaluation would be appropriate as it would allow to fix a very long standing issue (or at least provide a decent workaround).
